pkg/config/yaml.go

```diff
@@ -74,6 +74,26 @@ func importBlueprint(f string) (Blueprint, YamlCtx, error) {
 	return bp, yamlCtx, nil
 }
 
+func importDeploymentFile(f string) (DeploymentSettings, YamlCtx, error) {
```

Comment:
Can we some code sharing with `importBlueprint` above. Either by pulling out reading the yaml into a separate shared function, or by making the function generic and passing in the type that is being decoded.

Reply:
I gave this some thought when writing it. I do not see an obvious answer in Go. Passing in a pointer would not help as it would need two different types of pointers.

I think the main opportunity might be to introduce custom errors before decoding, e.g. FileNotFound and YAMLParsingError. I'm not sure there's an obvious net simplification.

@mr0re1 ?

Reply:
How about the following (passes tests):

```
func readYaml(f string) (*yaml.Decoder, YamlCtx, error) {
	data, err := os.ReadFile(f)
	if err != nil {
		return &yaml.Decoder{}, YamlCtx{}, fmt.Errorf("%s, filename=%s: %v", errMsgFileLoadError, f, err)
	}
	decoder := yaml.NewDecoder(bytes.NewReader(data))
	decoder.KnownFields(true)

	yamlCtx, err := NewYamlCtx(data)
	if err != nil { // YAML parsing error
		return &yaml.Decoder{}, yamlCtx, err
	}
	return decoder, yamlCtx, nil
}

func importBlueprint(f string) (Blueprint, YamlCtx, error) {
	decoder, yamlCtx, err := readYaml(f)
	if err != nil {
		return Blueprint{}, YamlCtx{}, err
	}

	var bp Blueprint
	if err = decoder.Decode(&bp); err != nil {
		return Blueprint{}, yamlCtx, parseYamlV3Error(err)
	}
	return bp, yamlCtx, nil
}

func importDeploymentFile(f string) (DeploymentSettings, YamlCtx, error) {
	decoder, yamlCtx, err := readYaml(f)
	if err != nil {
		return DeploymentSettings{}, YamlCtx{}, err
	}

	var depl DeploymentSettings
	if err = decoder.Decode(&depl); err != nil {
		return DeploymentSettings{}, yamlCtx, parseYamlV3Error(err)
	}
	return depl, yamlCtx, nil
}
```